javaparser-core/src/main/java/com/github/javaparser/resolution/types/ResolvedUnionType.java

```diff
@@ -22,23 +22,27 @@
 package com.github.javaparser.resolution.types;
 
 import java.util.*;
-import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.joining;
 
 /**
  * A union type is defined in java as list of types separates by pipes.
  *
  * @author Federico Tomassetti
  */
 public class ResolvedUnionType implements ResolvedType {
-    private List<ResolvedType> elements;
+    private final List<ResolvedType> elements;
 
-    public ResolvedUnionType(List<ResolvedType> elements) {
+    public ResolvedUnionType(Collection<ResolvedType> elements) {
```

Comment:
Is order really irrelevant here?

Reply:
I would say yes, it is irrelevant. Still, if you pass in a list, order will be retained (for printing purposes mostly)